singularity-eos/closure/mixed_cell_models.hpp

```diff
@@ -275,10 +275,29 @@ class PTESolverBase {
     // check for sanity.  basically checks that the input temperatures weren't garbage
     PORTABLE_REQUIRE(Tguess < params_.temperature_limit,
                      "Very large input temperature or temperature guess");
+
+    // JMM: To get a better guess for temperature such that the
+    // energies add up, we sometimes take one Newton step, and accept
+    // if it makes the temperature larger. Empirically, we find this
+    // avoids local saddle points that the solver can have trouble
+    // navigating.
+    auto newton_step = [&](Real T) {
+      Real dudt = 0;
+      Real usum = 0;
+      for (std::size_t m = 0; m < nmat; ++m) {
+        Real sie = eos[m].InternalEnergyFromDensityTemperature(rho[m], T, lambda[m]);
+        Real cv = eos[m].SpecificHeatFromDensityTemperature(rho[m], T, lambda[m]);
+        Real usum = rhobar[m] * sie;
+        dudt += rhobar[m] * cv;
+      }
+      return T - (usum - utotal_scale * uscale) / dudt;
+    };
+
```

Comment:
Wouldn't just checking `dudt < 0` be equivalent? In effect, you're stepping the guess towards the local optimum.

I suspect the effect here is doing a partial flattening of the energy surface - similar to something like [basin hopping](https://en.wikipedia.org/wiki/Basin-hopping).

Reply:
What do you mean by checking `dudt < 0`?

Reply:
Is this worth putting this behind a flag? It's conceivable that downstream codes may have techniques for generating guesses that could conflict.

Reply:
yeah, I'd be supportive of hiding it behind a flag.

Reply:
> What do you mean by checking `dudt < 0`?

You accept the new guess if `T` increases, so in effect you're checking if `dudt` is negative (unless i'm mistaken and the numerator can be negative?)

Reply:
The numerator can be either sign. Both because the current total energy can be either side of the target and because the total (or target) energy can be negative

Reply:
> Is this worth putting this behind a flag? It's conceivable that downstream codes may have techniques for generating guesses that could conflict.

Done.